src/main/java/org/opensearch/neuralsearch/search/util/NeuralSparseTwoPhaseUtil.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.util;
+
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.BoostQuery;
+import org.apache.lucene.search.Query;
+import org.opensearch.neuralsearch.query.NeuralSparseQuery;
+import org.opensearch.neuralsearch.query.NeuralSparseTwoPhaseParameters;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.rescore.QueryRescorer;
+import org.opensearch.search.rescore.RescoreContext;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.Float.max;
+
+/**
+ * Util class for do two phase preprocess for the NeuralSparseQuery.
+ * Include adding the second phase query to searchContext and set the currentQuery to highScoreTokenQuery.
+ */
+public class NeuralSparseTwoPhaseUtil {
+    /**
+     * @param query         The whole query include neuralSparseQuery to executed.
+     * @param searchContext The searchContext with this query.
+     */
+    public static void addRescoreContextFromNeuralSparseQuery(final Query query, SearchContext searchContext) {
+        Map<Query, Float> query2weight = new HashMap<>();
+        float windowSizeExpansion = populateQueryWeightsMapAndGetWindowSizeExpansion(query, query2weight, 1.0f, 1.0f);
+        Query twoPhaseQuery;
+        if (query2weight.isEmpty()) {
+            return;
+        } else if (query2weight.size() == 1) {
+            Map.Entry<Query, Float> entry = query2weight.entrySet().stream().findFirst().get();
+            twoPhaseQuery = new BoostQuery(entry.getKey(), entry.getValue());
+        } else {
+            twoPhaseQuery = getNestedTwoPhaseQuery(query2weight);
+        }
+        int curWindowSize = (int) (searchContext.size() * windowSizeExpansion);
+        if (curWindowSize < 0 || curWindowSize > NeuralSparseTwoPhaseParameters.MAX_WINDOW_SIZE) {
```

Comment:
how about change to curWindowSize <= 0?

Reply:
Actually, searchContext.size() is not a IllegalArgumentException on most query.
Such as, use a query with size=0 to get the return information which include hits and other metadata; and do a profile neuralsparse query with size=0 is also possible. So I think maybe there a little help to keep not throwing exception when curWindowsize=0, but make a explanation in doc?